content/100-getting-started/02-setup-prisma/100-start-from-scratch/110-relational-databases/100-connect-your-database.mdx

```diff
@@ -111,6 +111,84 @@ DATABASE_URL="mysql://root:randompassword@localhost:3306/mydb"
+
+```prisma file=prisma/schema.prisma highlight=2;edit
+datasource db {
+  provider = "mysql"
+  url      = env("DATABASE_URL")
+}
+```
+
+You will also need to add the [`referentialIntegrity` preview feature](/concepts/components/prisma-schema/relations/referential-integrity) to the `generator` block of your `schema.prisma` file, and set the referential integrity type to `"prisma"` in the `datasource` block:
+
+```prisma file=schema.prisma highlight=3;add|9;add
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["referentialIntegrity"]
+}
+
+datasource db {
+  provider             = "mysql"
+  url                  = env("DATABASE_URL")
+  referentialIntegrity = "prisma"
+}
+```
+
+The `url` is [set via an environment variable](/concepts/components/prisma-schema#accessing-environment-variables-from-the-schema) which is defined in `.env`:
+
+```bash file=.env
+DATABASE_URL="mysql://janedoe:mypassword@server.us-east-2.psdb.cloud/mydb?sslaccept=strict"
+```
+
+You now need to adjust the connection URL to point to your own database.
+
+The format of the connection URL for your database typically depends on the database you use. PlanetScale uses the MySQL connection URL format, which has the following structure (the parts spelled all-uppercased are _placeholders_ for your specific connection details):
+
+```no-lines
+mysql://USER:PASSWORD@HOST:PORT/DATABASE
+```
+
+Here's a short explanation of each component:
+
+- `USER`: The name of your database user
+- `PASSWORD`: The password for your database user
+- `PORT`: The port where your database server is running (typically `3306` for MySQL)
+- `DATABASE`: The name of the [database](https://dev.mysql.com/doc/refman/8.0/en/creating-database.html)
+
+For a database hosted with PlanetScale, the [connection URL](/reference/database-reference/connection-urls) looks similar to this:
+
+```bash file=.env
+DATABASE_URL="mysql://myusername:mypassword@server.us-east-2.psdb.cloud/mydb?sslaccept=strict"
+```
+
+The connection URL for a given database branch can be found from your PlanetScale account by going to the overview page for the branch and selecting the 'Connect' dropdown. In the 'Passwords' section, generate a new password and select 'Prisma' to get the Prisma format for the connection URL.
+
+Alternatively, you can connect to your PlanetScale database server using the [PlanetScale CLI](https://docs.planetscale.com/tutorials/automatic-prisma-migrations#:~:text=Install%20the-,PlanetScale%20CLI,-.), and use a local connection URL. In this case the connection URL will look like this:
+
+```bash file=.env
+DATABASE_URL="mysql://root@localhost:PORT/mydb"
+```
+
+To connect to your branch, use the following command:
+
+```terminal
+pscale connect prisma-test branchname --port PORT
+```
+
+The `--port` flag can be omitted if you are using the default port `3306`.
```

Comment:
Do we have collapsibles? If so, this would be a good candidate to remove from the default few.

Reply:
I've just moved this to a collapsible <details> section. Not sure how great this looks - thoughts?

Reply:
The design is bad in that it is missing padding before the collapsible - but nothing we can do about that.

From a usability perspective I think it might be good to add `Alternative to Connection String: `, `Alternative: ` or something like that in front of the "Connecting using the PlanetScale CLI" and it should be good.

Reply:
Done, that does look clearer.